securedrop_client/storage.py

```diff
@@ -196,37 +232,52 @@ def lazy_setattr(o: Any, a: str, v: Any) -> None:
 
 
 def update_sources(
-    remote_sources: List[SDKSource], local_sources: List[Source], session: Session, data_dir: str
+    remote_sources: List[SDKSource],
+    local_sources: List[Source],
+    skip_uuids: List[str],
+    session: Session,
+    data_dir: str
 ) -> None:
     """
-    Given collections of remote sources, the current local sources and a
+    Given collections of remote sources, the current local sources, a list of
+    UUIDs to skip, and a
     session to the local database, ensure the state of the local database
     matches that of the remote sources:
 
     * Existing items are updated in the local database.
     * New items are created in the local database.
+    * Items that have been flagged to skip are not re-created in the local database
+      (prevent re-downloading data that has just been locally deleted)
     * Local items not returned in the remote sources are deleted from the
       local database.
     """
     local_sources_by_uuid = {s.uuid: s for s in local_sources}
     for source in remote_sources:
         if source.uuid in local_sources_by_uuid:
-            # Update an existing record.
-            local_source = local_sources_by_uuid[source.uuid]
-            lazy_setattr(local_source, "journalist_designation", source.journalist_designation)
-            lazy_setattr(local_source, "is_flagged", source.is_flagged)
-            lazy_setattr(local_source, "interaction_count", source.interaction_count)
-            lazy_setattr(local_source, "document_count", source.number_of_documents)
-            lazy_setattr(local_source, "is_starred", source.is_starred)
-            lazy_setattr(local_source, "last_updated", parse(source.last_updated))
-            lazy_setattr(local_source, "public_key", source.key["public"])
-            lazy_setattr(local_source, "fingerprint", source.key["fingerprint"])
+
+            # If source UUID has been flagged as having been locally updated, wait
+            # til next sync to update it. The list of skip_uuids is managed by the
+            # parent method and purged after each sync.
+            if skip_uuids and source.uuid in skip_uuids:
+                logger.debug("Skip source update for {} (this sync only)")
```

Comment:
This is intentional, even though this PR is about safe delete - we don't want to update a source's file count, document count, etc., on the first sync if their files and messages have been deleted locally, since those values will be incorrect.